src/ArborX_LinearBVH.hpp

```diff
@@ -130,10 +148,11 @@ class BoundingVolumeHierarchy
   {}
 
   template <typename ExecutionSpace, typename Predicates, typename Callback>
-  void query(ExecutionSpace const &space, Predicates const &predicates,
-             Callback const &callback,
-             Experimental::TraversalPolicy const &policy =
-                 Experimental::TraversalPolicy()) const
+  std::enable_if_t<Kokkos::is_execution_space<ExecutionSpace>::value>
```

Comment:
Why did you add enable_if?

Reply:
Overload resolution picked the wrong overload.

Reply:
How is that so?  There is no ambiguity between the two.

Reply:
To elaborate why I care about this, arguments passed to the old BVH::query overload are checked via these `check_valid_*` function and the SFINAE would yield poor diagnostic and confuse users.